accepted/2021/statics-in-interfaces/README.md

```diff
@@ -6,8 +6,1267 @@
+
+### Base Interfaces
+
+The base interfaces each describe one or more core operations that are used to build up the core interfaces listed later. In many cases, such as for addition, subtraction, multiplication, and division, there is only a single operator per interface. This split is important in order for them to be reusable against downstream types, like `DateTime` where addition does not imply subtraction or `Matrix` where multiplication does not imply division. These interfaces are not expected to be used by many users and instead primarily by developers looking at defining their own abstractions, as such we are not concerned with adding simplifying interfaces like `IAddable<TSelf> : IAddable<TSelf, TSelf>`. Likewise, every interface added introduces increased size and startup overhead, so we need to strike a balance between expressiveness, exxtensibility, and cost.
+
+Some of the code below assumes that https://github.com/dotnet/csharplang/issues/4665 is going to be accespted into the language. If it is not accepted or if the approved syntax differs, the declarations will need to be updated.
+
+```csharp
+namespace System
+{
+    public interface IParseable<TSelf>
+        where TSelf : IParseable<TSelf>
+    {
+        // These do not take NumberStyles as not all parseable types are numbers (such as Guid)
+        // Instead, INumber<TSelf> exposes additional parse methods that do take NumberStyles
+
+        static abstract TSelf Parse(string s, IFormatProvider? provider);
+
+        static abstract bool TryParse(string s, IFormatProvider? provider, out TSelf result);
+    }
+
+    public interface ISpanParseable<TSelf> : IParseable<TSelf>
+        where TSelf : ISpanParseable<TSelf>
+    {
+        // This inherits from IParseable, much as ISpanFormattable is planned to inherit from IFormattable
```

Comment:
No, we don't strictly need both and could have just `IParseable` and have it expose the `Span` overloads.
* This might negatively impact other languages and has the general downsides of exposing just span overloads, however

I figured that is a question for API review, however. I also believe DIM support in v1 is still up in the air.

Reply:
> This might negatively impact other languages and has the general downsides of exposing just span overloads, however

From my perspective, so be it :smile:.  We're introducing a brand new feature here in static interfaces that no one currently supports, so to even hit an issue with spans you'd first need to update your language to support this large new capability.  At that point, supports spans, too, or at least support passing the result of `string.AsSpan()` instead of `string` as an argument.